include/boost/numeric/ublas/matrix.hpp

```diff
@@ -3436,8 +3437,14 @@ namespace boost { namespace numeric {
 #ifdef BOOST_UBLAS_ENABLE_PROXY_SHORTCUTS
         using matrix_container<self_type>::operator ();
 #endif
+#ifdef BOOST_NO_CXX11_ALLOCATOR
         typedef typename ALLOC::size_type size_type;
         typedef typename ALLOC::difference_type difference_type;
+#else
+        typedef std::allocator_traits<ALLOC> alloc_traits;
+        typedef typename alloc_traits::size_type size_type;
+        typedef typename alloc_traits::difference_type difference_type;
```

Comment:
We have `<boost/core/allocator_access.hpp>`  now which simplifies this. Instead of:

```
#ifdef BOOST_NO_CXX11_ALLOCATOR
typedef typename ALLOC::size_type size_type;
#else
typedef typename std::allocator_traits<ALLOC>::size_type size_type;
#endif 
```
This can be simply:
```
typedef typename boost::allocator_size_type<ALLOC>::type size_type;
```
It is our equivalent of `allocator_traits` just with individual traits and functions.